kernel/src/sched.rs

```diff
@@ -611,9 +611,7 @@ impl Kernel {
                     // underlying timer is not affected.
                     process.setup_mpu();
                     chip.mpu().enable_mpu();
-                    scheduler_timer.arm();
                     let context_switch_reason = process.switch_to();
-                    scheduler_timer.disarm();
```

Comment:
comment to note that I discussed this with Amit out of band, and this change should be fine now that we no longer rely on a global variable for detecting systick expirations.